packages/nimble-components/src/utilities/unit-format/manually-translated-unit-scale/base/manually-translated-scaled-unit-format.ts

```diff
@@ -0,0 +1,74 @@
+
+/**
+ * A formatter for units that are not supported by Intl.NumberFormat
+ */
+export class ManuallyTranslatedScaledUnitFormat extends ScaledUnitFormat {
+    private readonly pluralRules: Intl.PluralRules;
+    private readonly formatter: Intl.NumberFormat;
+    private readonly unitTranslation: UnitTranslation;
+
+    protected constructor(
+        scaledUnitFormatFactoryOptions: ScaledUnitFormatFactoryOptions,
+        private readonly unitTranslations: ReadonlyMap<string, UnitTranslation>,
+        private readonly unitPrefix: UnitPrefix
+    ) {
+        super(scaledUnitFormatFactoryOptions);
+        this.pluralRules = new Intl.PluralRules(this.locale);
+        this.formatter = new Intl.NumberFormat(
+            this.locale,
+            this.intlNumberFormatOptions
+        );
+        this.unitTranslation = this.getTranslationToUse(this.locale);
+    }
+
+    public static createFactory(
+        unitTranslations: ReadonlyMap<string, UnitTranslation>,
+        unitPrefix: UnitPrefix
+    ) {
+        return (
+            scaledUnitFormatFactoryOptions: ScaledUnitFormatFactoryOptions
+        ): ManuallyTranslatedScaledUnitFormat => new ManuallyTranslatedScaledUnitFormat(
+            scaledUnitFormatFactoryOptions,
+            unitTranslations,
+            unitPrefix
+        );
+    }
+
+    public format(value: number): string {
+        const formatted = this.formatter.format(value);
+        if (!this.unitPrefix.isBase()) {
+            return `${formatted} ${this.unitPrefix.text}${this.unitTranslation.symbol}`;
+        }
+
+        // Some languages have more than two forms (singular/plural) of cardinal
+        // numbers, but we are treating anything other than the 'one' form as plural.
+        // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/PluralRules#description
+        //
+        // Because PluralRules.select() takes a number (not a string), it cannot differentiate
+        // between 1 and 1.0. When NumberFormat is configured to format with a set number
+        // of fractional digits, those fractional digits can have an effect on the pluralization
+        // of the unit. E.g. in English, it formats "1 byte" vs "1.0 bytes". Thus there is
+        // sometimes an inconsistency between unit pluralization for the same number, based
+        // on whether it's supported by NumberFormat, or manually translated.
+        const unitLabel = this.pluralRules.select(value) === 'one'
+            ? this.unitTranslation.singular
+            : this.unitTranslation.plural;
+        return `${formatted} ${unitLabel}`;
+    }
+
+    private getTranslationToUse(locale: string): UnitTranslation {
+        const localeObject = new Intl.Locale(locale ?? '');
+        const language = localeObject.language;
+        const region = localeObject.region;
+        const languageAndRegion = `${language}-${region?.toUpperCase() ?? ''}`;
```

Comment:
Can you use `localeObject.toLocaleString()` here instead? It'd be good to avoid any custom logic about how locale strings are constructed.

For example right now I think this handles the region being upper or lower case but the language is assumed to be lower case. In my testing `Intl.Locale` handles that by normalizing `language` to lower, `region` to upper, and `toLocaleString()` to `lower-UPPER`.

Reply:
Using `toLocaleString()` does seem better, but it looks like what I had would have worked reliably anyway (since `Intl.Locale` already normalizes the values of `language` and `region`). The `toUpperCase()` call was apparently redundant.

Reply:
I had to revert back to the original logic (minus the unnecessary `toUpperCase()` call), because `localeObject.toLocaleString()` can include subtags from the input locale string that we don't care about (e.g. in "fr-Latn-CA-u-hc-h12", we don't care about anything other than "fr" and "CA"). In other words, if we have translations for "fr-CA", we want to use them whether the locale is "fr-CA" or "fr-Latn-CA" or "fr-CA-u-hc" or so on.